extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -89,6 +89,20 @@ public CompletionStage<V> get() {
                 });
     }
 
+    @Override
+    public <K, V> Uni<V> getIfPresent(K key) {
```

Comment:
Good points, I would try to elaborate:

* I was not aware of the possibility to cache the `null` value.. I think there should be a distinction between cached `null` and key not in the cache, thus I can propose one of the following options:
    1. Return `CompletableFuture` or `null` if suggested
    2. Return `Uni` or `null` same as above, although returning `Uni` that can be `null` sounds strange
    3. Not sure if there is any cache value wrapper data class in the Quarkus, that we can use to have `Uni<CachedValue<V>>` so that we can emit null in case no key exists and wrapper can have null or not null values..
* `+1` on Uni computed lazily (if we go this direction)

I am open to all suggestions, just let me know what's your preference.

Reply:
@gwenneg Still waiting for the answer here :arrow_up:

Reply:
Sorry, I'm on PTO and responding slowly 😄

I would start simple with option 1: return `CompletableFuture` or `null` depending on the availibility of the key in the cache. If there's a real need for the `Uni` signature in the future, it can always be added later.

Reply:
Agreed, will try to finalize this today.

Reply:
I am sorry that it took so long, but I finally managed to update the PR as requested.

However, are you sure that caching `null` values is possible? Because I added the test below, that seems to be failing and I would expect if caching null is possible that it would be green:

```
    @Test
    public void testCacheNullValue() throws Exception {
        // with custom key
        Object key = new Object();

        // when null is cached
        cache.get(key, (k) -> null);

        // assert
        assertKeySetContains(key);
        assertGetIfPresent(key, null);
    }
```

I mean the whole change to `CompletableFuture` was done in order to support `null` values.

Reply:
Cool thanks, sure I was missing something.. I'll include the test as well then, with caching the `NullValueConverter.toCacheValue(null)` :+1:

Reply:
I added the test, original test was having a bug :).. Works perfectly..